### PR TITLE
SAK-45531: Course grade override should show either exact percentage override or calculated percentage

### PR DESCRIPTION
**Pull request by victorGomollon**
https://jira.sakaiproject.org/browse/SAK-45531

Course grade override should show either exact percentage override or calculated percentage

### CODE REVIEW

**Comment by SedueRey** [comment]
This PR fixes the behaviour that T&L have requested but, also, keeps the behaviour S2U needed to fulfill our instructors requierements. No extra UI to avoid adding more complexity. 

**Of course**, default behaviour is what T&L have requested.

File: config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties [hunk 1]
```diff
@@ -1913,6 +1913,10 @@
 # Default: false
 # gradebook.coursegrade.displayed=true
 
+# Default for course grade calculated
+# Default: true
+# gradebook.coursegrade.isCalculatedGrade=true
```
Comment (jesusmmp) [inline config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties:1918]: This should be set to false

Comment (SedueRey) [reply config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties:1918]: Thanks, @jesusmmp

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java [hunk 2]
```diff
@@ -3020,4 +3020,7 @@ public SecurityAdvisor addSecurityAdvisor() {
 	public void removeSecurityAdvisor(final SecurityAdvisor advisor) {
 		this.securityService.popAdvisor(advisor);
 	}
+	public boolean getIsCalculatedGrade() {
```
Comment (ottenhoff) [inline gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java:3023]: this is a confusing method name. I would recommend showCalculatedGrade and getShowCalculatedGrade

Comment (SedueRey) [reply gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java:3023]: Of course.

File: config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties (hunk 1 above)
Comment (ern) [reply config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties:1918]: gradebook.coursegrade.showCalculatedGrade

**Comment by jesusmmp** [comment]
@ottenhoff Changes has been done! Please merge. Thanks

**Comment by ottenhoff** [comment]
If I enter "A", it now works as expected (previous behavior)

If I enter "90", it does not work as expected. I expect the "90" to stick!

**Comment by juanjmerono** [comment]
The original behavior in 19 was to change only the letter not the number. Changing the number is a NEW requirement, I mean that applying this patch implies that everything works exactly as before in 19 (or 20 reverting the original patch).

I think we should separate the work here, apply this patch (to solve the blocker now), and open a new ticket to change the number as required by T&L.

Right now the desired behavior of changing the number is not easy to achieve, as there is nothing to indicate whether the overwrite was done by changing the letter or the number, so executing this part of the job will take a little longer.

Sounds reasonable do it like that?

**Comment by ottenhoff** [comment]
Sure, but what have these JIRAs accomplished? You can now enter a number as an override but the numerical override displays as the lower bound of the corresponding letter grade? I'm wondering if it would be better to revert all and reconsider.

**Comment by juanjmerono** [comment]
Now? You could enter a number in Sakai 19 also, before the original patch was added, I mean in Sakai 19 you can override a grade using the number (p.e. 50) and you still view the original number (with the letter F). With this patch applied the behavior is exactly the same, but with a property that let us you change it.

In Sakai 19 WITHOUT ANY PATCH
Original Grade A+(100%)
You override an F => F(100%)
You override with 51 => F(100%)

In Sakai 20 WITH THIS PATCH (Default property value) OR reverting the patch
Original Grade A+(100%)
You override an F => F(100%)
You override with 51 => F(100%)

In Sakai 20 WITH THIS PATCH AND CHANGING THE PROPERTY (The behavior S2U needs)
Original Grade A+(100%)
You override an F => F(50%)
You override with 51 => F(51%)

In Sakai 20 With another patch (T&L requested behavior)
Original Grade A+(100%)
You override an F => F(100%)
You override with 51 => F(51%)

And yes if you override with a letter is using the corresponding lower bound.

**Comment by SedueRey** [comment]
Hello, @ottenhoff 

in sakai19, it didn’t work that way.

The current error is that this table contains the records of the items that are overwritten and it's final notes (which in java correspond to two different classes/entities).

Then the entity corresponding to the items, if it saved the values in the table in the column POINTS_EARNED (For the weight) and ENTERED_GRADE (for the scale).

But for the final grades we just saved the ENTERED_GRADE, the points one didn’t even exist, but we change it and used another attribute of the class and now can be used by the instructor.

In 19 the field in the final grade never were stored in database.

Also, we added a property for it to work in two different ways:
- One as it worked in 19 
- Another (with property) where you can overwrite it in two ways:
 · As a number, it keeps the number
 · As a letter, is overwritten as the lower numeric value of the letter.

**Review by ottenhoff — APPROVED** [review]
Okay thanks for the detailed explanation

**Pull request merged**